synapse/rest/client/v1/login.py

```diff
@@ -404,9 +404,11 @@ def __init__(self, hs):
     async def get_sso_url(
         self, request: SynapseRequest, client_redirect_url: bytes
     ) -> bytes:
-        return await self._oidc_handler.handle_redirect_request(
-            request, client_redirect_url
-        )
+        return (
+            await self._oidc_handler.handle_redirect_request(
+                request, client_redirect_url
+            )
+        ).encode("utf8")
```

Comment:
note that this was previously returning a `str` where a `bytes` was expected. (It worked ok because you can `request.redirect()` to either a `str` or a `bytes`, but it was violating the type annotation)